### PR TITLE
[raft] move updating atime logic to raft/cache

### PR DESCRIPTION
**Pull request by luluz66**
**Related issues**: N/A

### CODE REVIEW

**Review by tylerwilliams — APPROVED** [review]

File: enterprise/server/raft/cache/cache.go [hunk 1]
```diff
@@ -40,12 +42,15 @@ import (
...
-	clearCacheOnStartup = flag.Bool("cache.raft.clear_cache_on_startup", false, "If set, remove all raft + cache data on start")
-	partitions          = flag.Slice("cache.raft.partitions", []disk.Partition{}, "")
-	partitionMappings   = flag.Slice("cache.raft.partition_mappings", []disk.PartitionMapping{}, "")
+	rootDirectory        = flag.String("cache.raft.root_directory", "", "The root directory to use for storing cached data.")
+	httpAddr             = flag.String("cache.raft.http_addr", "", "The address to listen for HTTP raft traffic. Ex. '1992'")
+	gRPCAddr             = flag.String("cache.raft.grpc_addr", "", "The address to listen for internal API traffic on. Ex. '1993'")
+	clearCacheOnStartup  = flag.Bool("cache.raft.clear_cache_on_startup", false, "If set, remove all raft + cache data on start")
+	partitions           = flag.Slice("cache.raft.partitions", []disk.Partition{}, "")
+	partitionMappings    = flag.Slice("cache.raft.partition_mappings", []disk.PartitionMapping{}, "")
+	atimeUpdateThreshold = flag.Duration("cache.raft.atime_update_threshold", 10*time.Minute, "Don't update atime if it was updated more recently than this")
```
Comment (tylerwilliams) [inline enterprise/server/raft/cache/cache.go:51]: this seems p short

File: enterprise/server/raft/cache/cache.go [hunk 2]
```diff
@@ -72,6 +81,10 @@ type Config struct {
 	PartitionMappings []disk.PartitionMapping
 }
 
+type accessTimeUpdate struct {
```
Comment (tylerwilliams) [inline enterprise/server/raft/cache/cache.go:84]: nit: comment that this struct exists to put in the accesses chan

File: enterprise/server/raft/cache/cache.go [hunk 3]
```diff
@@ -546,3 +623,89 @@ func (rc *RaftCache) SupportsCompressor(compressor repb.Compressor_Value) bool {
 func (rc *RaftCache) SupportsEncryption(ctx context.Context) bool {
 	return false
 }
+
+func (rc *RaftCache) olderThanThreshold(t time.Time, threshold time.Duration) bool {
```
Comment (tylerwilliams) [inline enterprise/server/raft/cache/cache.go:627]: not sure this func adds much value (and it's only called once)

**Review by tylerwilliams — APPROVED** [review]

File: enterprise/server/raft/cache/cache.go (hunk 2 above)
Comment (luluz66) [reply enterprise/server/raft/cache/cache.go:85]: done

File: enterprise/server/raft/cache/cache.go (hunk 3 above)
Comment (luluz66) [reply enterprise/server/raft/cache/cache.go:628]: removed.

File: enterprise/server/raft/cache/cache.go (hunk 1 above)
Comment (luluz66) [reply enterprise/server/raft/cache/cache.go]: changed to 3 hours.

**Pull request merged**